src/pages/AddReservations.jsx

```diff
@@ -1,28 +1,126 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
 import backimg from '../assets/background.jpg';
 
+import { useGetAllVespasQuery, useCreateReservationMutation } from '../redux/vespaAPI';
+
 function AddReservations() {
+  // const data = ['vespa1', 'vespa2', 'vespa3', 'vespa4', 'vespa5', 'vespa6'];
+
+  const { data: vespas, error, isLoading } = useGetAllVespasQuery();
+
+  console.log(vespas);
+
+  const userID = useSelector((state) => state.persistedReducer.id);
+
+  const minDate = new Date().toISOString().slice(0, 10);
+
+  const [endDateMinDate, setEndDateMinDate] = useState(minDate);
+
+  const [startDateMaxDate, setStartDateMaxDate] = useState(null);
+
+  const [selectedVespa, setSelectedVespa] = useState('Select a Vespa');
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+
+  const [createReservation, { isLoading: isCreating }] = useCreateReservationMutation();
+
+  const handleDropDownChange = (e) => {
+    setSelectedVespa(e.target.value);
+  };
+
+  const handleStartDateChange = (e) => {
+    setStartDate(e.target.value);
+    setEndDateMinDate(e.target.value);
+  };
+
+  const handleEndDateChange = (e) => {
+    setEndDate(e.target.value);
+    setStartDateMaxDate(e.target.value);
+  };
+
+  const handleSubmitClick = () => {
+    const reservation = {
+      reservation: {
+        user_id: userID,
+        room_id: 1,
+        start_date: startDate,
+        end_date: endDate,
+        description: 'I made a reservation FINALLLYY!!',
+      },
+    };
```

Comment:
- please if this function is for testing, remove it to make your code clear, if it's not please notice the room_id is constant here 👓

Reply:
I have fixed this :)